.travis.yml

```diff
@@ -23,17 +25,17 @@ script:
 
 after_success:
   - codecov
+  - if [[ $BUILD_INSTALLER == 'yes' ]]; then pip install pynsist; ./script/makeinstaller.sh; fi
 
-before_deploy:
-  - sudo apt-get install nsis
-  - sudo pip install s3cmd
-  - pip install pynsist
-  - "./script/makeinstaller.sh"
+addons:
+  apt:
+    packages:
+    - nsis
 
 deploy:
   provider: script
   skip_cleanup: true
-  script: if [ -n "${TRAVIS_TAG}" ]; then s3cmd put build/nsis/streamlink-*.exe s3://streamlink-builds/release/windows/; else s3cmd put build/nsis/streamlink-*.exe s3://streamlink-builds/nightly/windows/; s3cmd put build/nsis/streamlink-*.exe s3://streamlink-builds/nightly/windows/streamlink-latest.exe; fi
+  script: sudo pip install s3cmd; if [ -n "${TRAVIS_TAG}" ]; then s3cmd put build/nsis/streamlink-*.exe s3://streamlink-builds/release/windows/; else s3cmd put build/nsis/streamlink-*.exe s3://streamlink-builds/nightly/windows/; s3cmd put build/nsis/streamlink-*.exe s3://streamlink-builds/nightly/windows/streamlink-latest.exe; fi
```

Comment:
`s3cmd` doesn't work with Python 3, and is only required when there is a deploy to S3, in this case I thought it better to install it only for the deploy. But, I don't mind moving it to  `before_install` with the others. I would have used the version in `apt` but it's a bit outdated and doesn't support loading the secret id/key from environment variables. 
The Travis S3 deploy wants to upload all of the files, I couldn't get it to upload only small set of files, also I wanted to to copy the current installer to `streamlink-latest.exe`.